### ISSUE TITLE
Document custom method parameter

### ISSUE THREAD

**Issue by NaikSoftware** [springfox/springfox#1683]
How to configure documentation for method parameter with custom HandlerMethodArgumentResolver?
For example:
``` java
@RequestMapping(value = '/user', method = RequestMethod.GET)
def getUser(@VersionApi int version,
                   @RequestParam('id') String id){
        // ...
}
```
`VersionApi` - custom annotation that handled by my `HandlerMethodArgumentResolver` implementation.

Now version showed as ParameterType "body". I want show as "header".

**Comment by dilipkrish** [comment]
TL;DR is that you would have to build your own plugin to do it.

Basically the only out-of-the-box annotations to augment the description in this case are `@ApiParam` and to be more accurate `@ApiImplicitParam`. Unfortunately neither of those annotations support descriptions.

So my suggestion would be to:
 
1. Create your own annotation that would look like this

    `@VersionApi String apiKey`

NOTE: There is already an [annotation in spring][1] that is suitable for this.

2. Create your own [ParameterBuilderPlugin][2]
3. Implement the plugin as shown below



    public class VersionApiReader implements ParameterBuilderPlugin {
      @Override
      public void apply(ParameterContext parameterContext) {
        ResolvedMethodParameter methodParameter =parameterContext.resolvedMethodParameter();
        Optional<VersionApi> requestParam = methodParameter.findAnnotation(VersionApi.class);
        if (requestParam.isPresent()) {
          parameterContext.parameterBuilder()
                          .parameterType("header)
                          .parameterType("id");
        }
      }

      @Override
      public boolean supports(DocumentationType documentationType) {
        return false;
      }
    }

4. Pick a value of [the order][3] that is is applied [after swagger annotations][4] have been processed.

  [1]: http://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Description.html
  [2]: https://github.com/springfox/springfox/blob/master/springfox-spring-web/src/main/java/springfox/documentation/spring/web/readers/parameter/ParameterRequiredReader.java#L45
  [3]: https://github.com/springfox/springfox/blob/master/springfox-spring-web/src/main/java/springfox/documentation/spring/web/readers/parameter/ParameterRequiredReader.java#L44
  [4]: https://github.com/springfox/springfox/blob/master/springfox-swagger-common/src/main/java/springfox/documentation/swagger/common/SwaggerPluginSupport.java#L30
  [5]: http://mvnrepository.com/artifact/io.springfox/springfox-swagger2
